### PR TITLE
Use ethers which supports better providers out of the box

### PR DESCRIPTION
**Pull request by ziogaschr**
This allows "http://", "ws://" and "*.ipc" to be used for reaching the provider.

### CODE REVIEW

**Comment by ziogaschr** [comment]
@meowsbits @diega what do you think if we accept the port or path to file to be passed through the path? This way we can run multiple clients on a single machine.

Like: 
- "http://<proxy_api_host>/http/<port>"
- "http://<proxy_api_host>/ws/<port>"
- "http://<proxy_api_host>/ipc/<filename>"

p.s. on IPC filename will be passed and not the path, which will be hardcoded in config. This will be forced in order they can't search in our filesystem.

**Comment by ziogaschr** [comment]
This branch is already deployed at bootnodes-classic-sfo3-01

**Review by meowsbits — APPROVED** [review]

**Comment by meowsbits** [comment]
> what do you think if we accept the port or path to file to be passed through the path? This way we can run multiple clients on a single machine.

I think we should keep this program as simple as possible. If we add routing for multiple providers, we'll need a way to define the configuration for that, and the program will have to assume some kind of opinion/convention for namespacing/patterning the listening URIs. And we'll have to figure out how to deal with potential errors from multiple sources (do we exit if _any_ provider fails? or if they _all_ fail?). Can each provider get their own WHITELIST too? 

I think we should leave routing to routing applications, like nginx or caddy. This way we'll run multiple instances simply with multiple instances:

```
env PORT=3000 CLIENT_PROVIDER_URL=/geth1/geth.ipc jsonrpc-gatekeeper 
env PORT=3001 CLIENT_PROVIDER_URL=/geth2/geth.ipc jsonrpc-gatekeeper 
```

and then have arbitrary choices for routing with a dedicated router:

eg. Caddy
```
ws1.rpc.etccooperative.org {
    reverse_proxy localhost:3000
}
ws2.rpc.etccooperative.org {
    reverse_proxy localhost:3001
}
```

or 

```
ws.rpc.etccooperative.org/0 {
    reverse_proxy localhost:3000
}
ws.rpc.etccooperative.org/1 {
    reverse_proxy localhost:3001
}
```

or...  any other pattern we want.

**Comment by ziogaschr** [comment]
I thought that you had a preference on a single instance running from what we discussed in the call. Either way, the client that calls these APIs have to know the ports to call.

```
rpc.etccooperative.org:3000/http/8545
rpc.etccooperative.org:3000/ws/8546
rpc.etccooperative.org:3000/ws/8547
```

**Comment by meowsbits** [comment]
> I thought that you had a preference on a single instance running

I did, and now I don't think so anymore :face_in_clouds:. 
The "single instance" limit is for a public webserver on the machine, ie. any (one) thing using port 443 or 80.

> Either way, the client that calls these APIs have to know the ports to call.

Not sure I follow?...

For example, for:

```
rpc.etccooperative.org:3000/http/8545
rpc.etccooperative.org:3000/ws/8546
rpc.etccooperative.org:3000/ws/8547
```

... you can do:

```
env PORT=3001 CLIENT_PROVIDER_URL=http://localhost:8545 jsonrpc-gatekeeper 
env PORT=3002 CLIENT_PROVIDER_URL=ws://localhost:8546 jsonrpc-gatekeeper 
env PORT=3003 CLIENT_PROVIDER_URL=ws://localhost:8547 jsonrpc-gatekeeper 
```

```
http://rpc.etccooperative.org:3000 {
    handle /http/8545 { reverse_proxy localhost:3001 }
    handle /ws/8546 { reverse_proxy localhost:3002 }
    handle /ws/8547 { reverse_proxy localhost:3003 }
}
```

**Comment by meowsbits** [comment]
Although maybe I could see something like `http://rpc.etccooperative.org:3000//geth/geth1/geth.ipc` or `https://rpc.etccooperative.org:3000/http://localhost:8545`, where you pass the provider URL directly to jsonrpc-gatekeeper for an on-demand multi-provider reverse-proxy...

File: .env.example [hunk 1]
```diff
@@ -4,8 +4,13 @@ PORT=3000
 # Systemd service to restart if the check fails
 SYSTEMD_SERVICE="proxied_rpc.service"
```
Comment (meowsbits) [inline .env.example]: PS. Maybe this should be removed? And _health_check.sh_ too...

Comment (ziogaschr) [reply .env.example]: Probably it can yes, but I wanted to keep it for a few days using ethers, just as a fail safe. Until we check how well IPC works

**Comment by ziogaschr** [comment]
Yes, both are exactly the same at the end.

> Although maybe I could see something like http://rpc.etccooperative.org:3000//geth/geth1/geth.ipc or https://rpc.etccooperative.org:3000/http://localhost:8545, where you pass the provider URL directly to jsonrpc-gatekeeper for an on-demand multi-provider reverse-proxy...

I don't find this as a good idea as it exposes too much information.

So, let's keep the single path this PR already has.

**Comment by meowsbits** [comment]
:kiss: 

Happy to wait til we figure something out in Ansible too.

**Pull request merged**